### PR TITLE
Specify binding ports for authentication

### PR DESCRIPTION
**Pull request by johnweldon**
Provide mechanism to specify binding port for authentication, addressing #229, using the context and a new `sasl.WithBindingPort(ctx context.Context, port uint16) ctx.Context` helper function and related code.

### CODE REVIEW

**Comment by johnweldon** [comment]
@jdef is there anyone you know of who is able to review and merge this?

**Comment by jdef** [comment]
@johnweldon i'll take a look. have been on PTO the last 10 days

File: .travis.yml [hunk 1]
```diff
@@ -6,7 +6,6 @@ go:
   - 1.5.1
   - tip
 before_install:
-  - go get golang.org/x/tools/cmd/vet
```
Comment (jdef) [inline .travis.yml]: how is this change related to this PR?

Comment (johnweldon) [inline .travis.yml]: The travis build failed because that repo no longer works.  I wouldn't put it in the PR, but that's the only way it got past travis.

**Comment by jdef** [comment]
LGTM. Thanks for the PR!

**Comment by johnweldon** [comment]
Thanks @jdef
